design/sketch.webidl

```diff
@@ -512,23 +514,23 @@ enum GPUInputStepMode {
 };
 
 dictionary GPUVertexAttributeDescriptor {
-    u32 shaderLocation;
-    u32 inputSlot;
-    u64 offset;
-    GPUVertexFormat format;
+    required u32 shaderLocation;
+    required u32 inputSlot;
+    u64 offset = 0;
+    required GPUVertexFormat format;
 };
 
 dictionary GPUVertexInputDescriptor {
-    u32 inputSlot;
-    u64 stride;
-    GPUInputStepMode stepMode;
+    required u32 inputSlot;
+    required u64 stride;
+    GPUInputStepMode stepMode = "vertex";
 };
 
 dictionary GPUInputStateDescriptor {
-    GPUIndexFormat indexFormat;
+    required GPUIndexFormat indexFormat;
```

Comment:
Can this be optional (throw an error if drawIndexed is called and this is not set)? What if drawIndexed is never called with this pipeline?

Reply:
I think it's worth considering; we ran into this issue when we were writing demos too. Let's do it in another issue/PR after this one, though.